File: 1945-sum-of-digits-of-string-after-convert/1945-sum-of-digits-of-string-after-convert.cpp
class Solution {
public:
    int getLucky(string s, int k) {
        int sum=0;
        
        for(auto a:s)
        {
            int val=(a-'a')+1;
            while(val)
            {
                sum+=val%10;
                val/=10;
            }
        }
        
        while(--k)
        {
            int r=0;
            while(sum)
            {
                r+=sum%10;
                sum/=10;
            }
            sum=r;
        }
        
        return sum;
    }
};

File: 1945-sum-of-digits-of-string-after-convert/1945-sum-of-digits-of-string-after-convert.java
class Solution {
    public int getLucky(String s, int k) {
      String n="";
        int sum=0;
        
        for(int i=0; i<s.length(); i++)
        {
            n+=s.charAt(i)-'a'+1+"";
        }
        while(k-->0)
        {
           sum=0;
            for(char c:n.toCharArray())
            {
                sum+=Integer.parseInt(c+"");
            }
            n=String.valueOf(sum);
        }
        return sum;
    }
}